examples/step-1/CMakeLists.txt

```diff
@@ -34,6 +35,5 @@ if(NOT ${deal.II_FOUND})
     )
 endif()
 
-deal_ii_initialize_cached_variables()
-project(${TARGET})
-deal_ii_invoke_autopilot()
+add_executable(${TARGET} ${TARGET_SRC})
+target_link_libraries(${TARGET} dealii::dealii)
```

Comment:
Would you mind to drop all changes to the example steps for the time being?
I would like to address this in a separate pull request.

In particular, I am not happy with this change:

 I think it is important that we still support the quirky setup in our example step. Your change will break our compiler autoselection and will remove the `debug`, `release`, `run` and cleanup targets. Therefore, I really think that we should keep the `deal_ii_invoke_autopilot()` macro around.